### PR TITLE
Add bash/zsh autocompletion

### PR DESCRIPTION
**Pull request by tetafro**
Issue https://github.com/birdayz/kaf/issues/14

I've added bash autocompletion and fixed some minor issues.

> note that it can be slow to get the list of topics. maybe add caching?

Not sure if this is a good idea. How would you manage such cache? What user should do to update his list of topics?

### CODE REVIEW

File: cmd/kaf/consume.go [hunk 1]
```diff
@@ -68,12 +68,11 @@ func getAvailableOffsetsRetry(
 }
 
 const (
-	offsetsRetry       = 500 * time.Millisecond
-	configProtobufType = "protobuf.type"
```
Comment (tetafro) [inline cmd/kaf/consume.go]: dead code

File: cmd/kaf/group.go [hunk 2]
```diff
@@ -256,13 +259,7 @@ var groupLsCmd = &cobra.Command{
 			fmt.Fprintf(w, "%v\t%v\t%v\t\n", detail.GroupId, state, consumers)
 		}
 
-		if len(groupDescs) != 0 {
-			w.Flush()
-		} else {
-			fmt.Printf("No Groups found\n")
-		}
```
Comment (tetafro) [inline cmd/kaf/group.go]: This was the only command, that returns message string if nothing is found. It was a problem for parsing output, so I just deleted it.

File: cmd/kaf/node.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 
 func init() {
 	rootCmd.AddCommand(nodeCommand)
+	rootCmd.AddCommand(nodesCommand)
```
Comment (tetafro) [inline cmd/kaf/node.go]: Lost command

**Comment by birdayz** [comment]
i don't quite understand how this exactly works. how would a user use it? i suppose running kaf completion and source it? i failed to test it to be honest.

**Comment by tetafro** [comment]
> i don't quite understand how this exactly works. how would a user use it? i suppose running kaf completion and source it? i failed to test it to be honest.

Yeah, you're right, I should've added some docs. Please check readme now (or tell me where to add this info).

**Comment by birdayz** [comment]
i have trouble getting this to work..
i put the required script into bashrc, then i typed: `kaf gr` and hit tab.

result:
```
j0e  joe-xps  ~  projects  kaf  tetafro-autocompletion  5⚑  $  kaf grobash: _get_comp_words_by_ref: command not found
```

**Comment by josher19** [comment]
Perhaps use compatibility mode:
```
autoload bashcompinit
bashcompinit
```
https://computingforgeeks.com/how-to-source-bash-completion-script-file-in-zsh/

However, looks like `type -t` will need to be changed and still might not work for zsh

Also looks like you should merge in the latest master branch.

**Comment by tetafro** [comment]
Cobra package can generate native zsh completion, I'll add an option to choose between bash and zsh.

File: cmd/kaf/completion.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+			rootCmd.GenBashCompletion(os.Stdout)
+		case "zsh":
+			if err := genZshCompletion(os.Stdout); err != nil {
+				panic(err)
+			}
+		}
+	},
+}
+
+// GenZshCompletion generates a zsh completion and writes to the passed writer.
```
Comment (tetafro) [inline cmd/kaf/completion.go]: Updated.

Sad, but seems like zsh autocompletion doesn't work in cobra for now. I used this hack to make it work.

**Comment by birdayz** [comment]
awesome. thank you !

**Pull request merged**